osquery/tables/ipmi/ipmi_info.cpp

```diff
@@ -0,0 +1,1297 @@
+
+  if (domain_ != nullptr) {
+    up_.store(true);
+  } else {
+    LOG(WARNING) << "IPMI BMC was not responsive";
+    return;
+  }
+
+  TLOG << "Searching for IPMI channel for LAN communication";
+  findLANCh();
+
+  TLOG << "Setting up ipmi_lanparm_t for system MCs";
+  rv = ipmi_domain_iterate_mcs(
+      domain_,
+      [](ipmi_domain_t* domain, ipmi_mc_t* mc, void* data) {
+        IPMIClient* c = reinterpret_cast<IPMIClient*>(data);
+        c->addLANParm(getMCKey(mc), mc);
+      },
+      this);
+  if (rv != 0) {
+    LOG(ERROR) << "Could not register ipmi_domain_iterate_mcs: "
+               << strerror(rv);
+  }
+
+  blkAndOp([]() { return false; }, 1000);
+
+  TLOG << "IPMIClient initialization complete";
+}
+
+ipmi_lanparm_t* IPMIClient::getLANParm(const std::string& name, ipmi_mc_t* mc) {
+  if (lanparms_.find(name) != lanparms_.end()) {
+    return lanparms_[name].get();
+  }
+
+  if (addLANParm(name, mc)) {
+    return lanparms_[name].get();
+  }
+
+  return nullptr;
+}
+
+bool IPMIClient::addLANParm(const std::string& name, ipmi_mc_t* mc) {
+  ipmi_lanparm_t* lp = nullptr;
+
+  int rv = ipmi_lanparm_alloc(mc, lanCh_, &lp);
+  if (rv != 0) {
+    LOG(ERROR) << "Could not allocate ipmi_lanparm_alloc for local BMC: "
+               << strerror(rv);
+    return false;
+  }
+
+  if (lp == nullptr) {
+    LOG(ERROR) << "Could not successfully allocation "
+                  "ipmi_lanparm_t*: still a nullptr";
+    return false;
+  }
+
+  lanparms_[getMCKey(mc)] =
+      std::unique_ptr<ipmi_lanparm_t, std::function<void(ipmi_lanparm_t*)>>(
+          lp, kFreeLANParm);
+  return true;
+}
+
+void IPMIClient::insertRowsQueue(std::string key,
+                                 const std::string& columnName,
+                                 const std::string& columnValue) {
+  rowsQueue_[key][columnName] = columnValue;
+}
+
+void IPMIClient::insertRowsQueue(const std::string& key, const Row& row) {
+  rowsQueue_[key] = row;
+}
+
+void IPMIClient::toQueryData(QueryData& results) {
+  for (const auto& mcData : rowsQueue_) {
+    results.push_back(mcData.second);
+  }
+}
+
+void IPMIClient::pushParmData(parmData* parm) {
+  WriteLock lock(parmMutex_);
+  parmsOnHeap_.push_back(parm);
+}
+
+void IPMIClient::rmParmData(parmData* parm) {
+  WriteLock lock(parmMutex_);
+  // Check if parm exists first.
+  auto i = std::find(parmsOnHeap_.begin(), parmsOnHeap_.end(), parm);
+  if (i == parmsOnHeap_.end()) {
+    return;
+  }
+
+  std::unique_ptr<parmData, std::function<void(parmData*)>> _(
+      parm, [&](parmData* p) {
+        delete p;
+        parmsOnHeap_.erase(i);
+      });
+}
+
+void IPMIClient::blkAndOp(std::function<bool()> ready, int timeoutDurMS) {
+  std::atomic<bool> done(false);
+
+  auto fut = std::async(std::launch::async, [&, this] {
+    while (timeoutDurMS > 0 && !done.load()) {
+      pauseMilli(5);
+      if (interrupted()) {
+        return;
+      }
+
+      timeoutDurMS -= 6;
```

Comment:
None in particular. If the constant `6` is what works best maybe it's fine to stick with it? It looked like it could've potentially was supposed to have been a `5` is all. I think the impression I got throughout the base was it seemed to be quite a few constant values that could potentially have been abstracted into variables.